control-plane/connect-inject/controllers/endpoints/endpoints_controller.go

```diff
@@ -119,6 +120,10 @@ type Controller struct {
 	// to Consul client agents.
 	EnableAutoEncrypt bool
 
+	// EnableTelemetryCollector controls whether the proxy service should be registered
+	// with config to enable telemetry forwarding.
+	EnableTelemetryCollector bool
```

Comment:
Is this supposed to be toggleable on a per pod basis through annotations?  Most of our features allow you to turn a feature on globally through the helm chart or per pod using annotations. 

An example, is how we check if metrics is enabled on the annotation (https://github.com/hashicorp/consul-k8s/blob/main/control-plane/connect-inject/metrics/metrics_configuration.go#L72)

Reply:
Thats an interesting idea. I guess if I was operating a cluster I wouldn't want consumers to be able to disable telemetry. We currently assume the feature is either enabled or disabled cluster wide.